File: LeetCode/2. Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* l3 = new ListNode();
        ListNode* start = l3;
        int ones = 0, carry = 0, sum = 0;
        
        while(l1 != nullptr || l2 != nullptr) {
            if (l1==nullptr) {
                l1 = new ListNode(0, nullptr);
            }
            if (l2==nullptr) {
                l2 = new ListNode(0, nullptr);
            }
            
            sum = l1->val + l2->val + carry;
            
            ones = int(sum % 10);
            carry = int(sum / 10);
            
            l3->val = ones;
            
            l1 = l1->next;
            l2 = l2->next;
            
            if (l1 || l2 || carry == 1) {
                l3->next = new ListNode();
                l3 = l3->next;
            }
        }
        
        if(carry == 1) {
            l3->val = 1;
        }
        
        return(start);
    }
};

File: LeetCode/2. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        carry = 0
        begin = ListNode()
        l3 = begin
        
        while(l1 or l2):
            if l1 is None:
                l1 = ListNode()
                
            if l2 is None:
                l2 = ListNode()
            
            total = l1.val + l2.val + carry
            
            l3.val = int(total % 10)
            carry = int(total / 10)
            
            l1 = l1.next
            l2 = l2.next
            
            if l1 or l2 or carry == 1:
                l3.next = ListNode()
                l3 = l3.next
            
        if carry == 1:
            l3.val = 1
            l3.next = None
            
        return begin